### PR TITLE
refactor(TaskManager): make task manager generic

### LINKED ISSUE #293: Refactor Task Manager and Task Factory for Improved Decoupling
### Current Limitation
TaskManager and TaskFactory are currently hardcoded with domain services like FormService. This makes the TaskManager non-reusable across different projects and violates the Open-Closed Principle. Adding new task types (like PaymentTask) requires modifying core task management components.

### Suggested Improvement
Refactor TaskFactory into a generic registry that accepts 'Builder' functions (constructors). This allows registering task types with their specific dependencies during the application bootstrap phase, keeping TaskManager and TaskFactory agnostic of business-specific services.

### Version
v0.1.0-dev

### Additional Context
This refactoring will use the 'Builder' approach where the TaskFactory stores a map of task types to constructor functions. This allows for clean dependency injection at the composition root (bootstrap) without changing the core plugin execution logic.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Refactor `TaskManager` to decouple it from the `FormService` and make it more generic. This is achieved by introducing a `TaskFactory` that handles the creation of task execution units, simplifying the `TaskManager`'s dependencies and improving maintainability.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- **Decoupled `TaskManager` from `FormService`**: Removed the direct dependency on `FormService` within the `taskManager` struct and its constructor.
- **Introduced `TaskFactory`**: The `TaskManager` now accepts a `plugin.TaskFactory` interface, which is responsible for building task executors.
- **Updated Bootstrap Logic**: Modified `backend/internal/app/bootstrap/app.go` to initialize the `TaskFactory` with the required configuration and services before passing it to `NewTaskManager`.
- **Simplified `taskManager` struct**: Removed the `config` field from the `taskManager` struct as it is now managed by the factory.
- **Fixed Tests**: Updated `backend/internal/task/manager/manager_test.go` to reflect the signature changes in `NewTaskManager`.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #293

## Screenshots/Demo

N/A

## Additional Notes

These changes are part of a broader effort to make the task management system more modular and easier to extend with new task types without modifying the core manager.

## Deployment Notes

N/A

### CODE REVIEW

File: backend/internal/task/manager/manager_test.go [hunk 1]
```diff
@@ -168,8 +165,8 @@ func TestInitTask(t *testing.T) {
 		// Pre-populate cache
 		mockPlugin.On("Init", mock.Anything).Return().Once()
 
-		container := container.NewContainer(taskID, uuid.NewString(), uuid.NewString(), plugin.InProgress, nil, nil, nil, mockPlugin, nil)
-		tm.containerCache.Set(taskID, container)
+		newContainer := container.NewContainer(taskID, uuid.NewString(), uuid.NewString(), plugin.InProgress, nil, nil, nil, mockPlugin, nil)
+		tm.containerCache.Set(taskID, newContainer)
```
Comment (sthanikan2000) [inline backend/internal/task/manager/manager_test.go]: Why you're renaming the variables to newContainer?
